scylla/src/statement/prepared_statement.rs

```diff
@@ -126,6 +126,10 @@ impl PreparedStatement {
     /// it assumes that all partition key columns are passed in values.
     /// Partition keys have specific serialization rules.
     /// Ref: <https://github.com/scylladb/scylla/blob/40adf38915b6d8f5314c621a94d694d172360833/compound_compat.hh#L33-L47>
+    ///
+    /// Note: this is no longer required to compute a token. This is because partitioners are now stateful and stream-based,
+    /// so they do not require materialised partition key.
+    /// The function is solely kept for backward compatibility purposes.
```

Comment:
I'm not sure that the only purpose of this function is to compute the token.

I'd change this paragraph so that it only explains that if one wants to calculate the token, there are more efficient methods than computing partition key and then using it to compute the token.